controller/composite/controller.go

```diff
@@ -298,7 +298,7 @@ func (pc *parentController) resolveControllerRef(childNamespace string, controll
 		// (except for namespaced child -> cluster-scoped parent).
 		parentNamespace = childNamespace
 	}
-	parent, err := pc.parentInformer.Lister().Get(parentNamespace, controllerRef.Name)
+	parent, err := common.GetParent(pc.parentInformer, parentNamespace, controllerRef.Name)
```

Comment:
Trying to understand the motive here.
Even if we expose the logic as a pure function, we need to name the function appropriately.
In this particular case, I felt Get from a lister provides enough info to the maintainer on how the particular instance is extracted. 

_AFAIK K8s API based Get is costlier than Informer's Get._

Reply:
So, in previous metacontroller implementation, lister was able to recognize if object was namespaced or not, and return object. In `client-go` Lister's implementation :
* if object is cluster scope, use `Lister.Get(name)`
* if object is namespaced, use `Lister.Namespace(ns).Get(name)`

So I extracted part of code which recognizes if object is namespaced to `common.GetParent` (quick and dirty - agree :) ) to avoid writing this in every place.


I will try to make it part of `dynamicinformer.ResourceInformer` and name appropriately.

Reply:
(also, I am trying in next PR to replacemetacontrollers `dynamic.informer` package in favor of `client-go` `k8s.io/client-go/dynamic/dynamicinformer` package but it will require more work, thus separation )